src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -324,6 +326,18 @@ protected AnalyzerDriver(ImmutableArray<DiagnosticAnalyzer> analyzers, AnalyzerM
             return builder.ToImmutable();
         }
 
+        private ImmutableDictionary<DiagnosticAnalyzer, ImmutableArray<CompilationAnalyzerAction>> MakeCompilationActionsByAnalyzer()
+        {
+            var builder = ImmutableDictionary.CreateBuilder<DiagnosticAnalyzer, ImmutableArray<CompilationAnalyzerAction>>();
+            var actionsByAnalyzers = this.analyzerActions.CompilationActions.GroupBy(action => action.Analyzer);
```

Comment:
if you use GroupBy, you probably can do ToImmutableDictioary(kv => kv.Key, kv.ToImmutableArray()) ?